### DISCUSSION TITLE
Google Cloud Installation -- 502 Bad Gateway

### DISCUSSION THREAD

**Discussion by tstahov** [NightscoutFoundation/xDrip#3728] [General]
I am trying to install Nightscout on the Google Cloud platform. When I try to connect to the URL with my web browser I get a 502 Bad Gateway error. 

I am wondering if this has to do with the OS version I used for my VM. The instructions indicated that I should use Ubuntu 20.04 LTS **Minimal** as the OS version, but that was not an available choice when I created the VM. Instead I used Ubuntu 20.04 LTS  without the "Minimal" specification.  Initially the bootstrap script aborted because of the wrong OS version. I commented out the OS test and reran the bootstrap script. Here is the part of the script I changed:

```
 if [[ ! "$ubversion" = "20.04"* ]] || [[ ! "$(which vi)" = "" ]] # If the selected version of ubuntu is not what we want or if the main version has been installed instead of minimal                                                     
  then
  clear
#  dialog --colors --msgbox "       \Zr Developed by the xDrip team \Zn\n\n\                                                                                                                                                                 
#The Ubuntu version on the virtual machine is incorrect.  You need to delete the virtual machine and create a new one.  Please refer to the #guide for the details." 10 50
#  exit
  fi

```
Here is what I get from "lsb_release -a"
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.6 LTS
Release:        20.04
Codename:       focal

I have included a screenshot of the status below and I don't see any missing modules. 

I have been able to use the "Edit variables using a web browser" function so I am able to connect to the web server on the VM. 

I will be using Dexcom Share for the data source. 

I have tried rerunning both the Install Nightscout Phase I and Phase II, but this didn't fix anything. 

Thank you for any suggestions you are able to provide.

![image](https://github.com/user-attachments/assets/9f9c19d8-f4da-457f-a59c-8e06e1753b15)

**Comment by Navid200** [comment]
You will be charged for that because it does not satisfy the free tier requirements.  You have selected more RAM than allowed for free tier.

With respect to the OS, I am going to check now.  It is highly unlikely that version of OS is not available.  If it is not, I will respond.

Otherwise, please follow the instructions.
Please do not modify the code.

**Reply by Navid200** [reply]
It's right there.  You just need to scroll down to see it.
I have even included video clips showing every step including scrolling down to see the minimal versions.

**Reply by tstahov** [reply]
Thanks.  I had assumed that the versions would be grouped by version number and so I didn't scroll down far enough.

**Reply by tstahov** [reply]
I just wanted to follow up and thank you for the help. Everything is working flawlessly. Thanks for developing and maintaining this software. It is amazing.

**Reply by Navid200** [reply]
You're welcome.
Thanks for the feedback
